tasks/main.yml

```diff
@@ -11,6 +11,21 @@
     state: "{{ docker_package_state }}"
   notify: restart docker
 
+- name: Ensure /etc/docker/ directory exists.
+  file:
+    path: /etc/docker
+    state: directory
+    mode: 0755
+  when: docker_daemon_options is not none
```

Comment:
Along with the change above, I _believe_ this would be `when: docker_daemon_options.keys() | length > 0` — along with changing the `when` in the next clause.